src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddConstraintValidatorsPass.php

```diff
@@ -26,22 +27,13 @@ public function process(ContainerBuilder $container)
         $validators = array();
         foreach ($container->findTaggedServiceIds('validator.constraint_validator') as $id => $attributes) {
             if (isset($attributes[0]['alias'])) {
-                $validators[$attributes[0]['alias']] = $id;
+                $validators[$attributes[0]['alias']] = new Reference($id);
             }
 
             $definition = $container->getDefinition($id);
-
-            if (!$definition->isPublic()) {
-                throw new InvalidArgumentException(sprintf('The service "%s" must be public as it can be lazy-loaded.', $id));
-            }
-
-            if ($definition->isAbstract()) {
```

Comment:
Seems like `ContainerBuilder::findTaggedServiceIds()` deserves a new argument ;) 
I can't do it this week so I opened https://github.com/symfony/symfony/issues/21761 in case someone is interested.